cmd/aws/create_test.go

```diff
@@ -0,0 +1,402 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	ssmTypes "github.com/aws/aws-sdk-go-v2/service/ssm/types"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockCredentialsProvider struct {
+	creds aws.Credentials
+	err   error
+}
+
+func (m *mockCredentialsProvider) Retrieve(ctx context.Context) (aws.Credentials, error) {
+	return m.creds, m.err
+}
+
+func TestValidateCredentials(t *testing.T) {
+	tests := []struct {
+		name        string
+		creds       aws.Credentials
+		err         error
+		expectedErr error
+	}{
+		{
+			name: "valid credentials",
+			creds: aws.Credentials{
+				AccessKeyID:     "test-access-key-id",
+				SecretAccessKey: "test-secret-access-key",
+				SessionToken:    "test-session-token",
+			},
+			err:         nil,
+			expectedErr: nil,
+		},
+		{
+			name:        "failed to retrieve credentials",
+			creds:       aws.Credentials{},
+			err:         errors.New("failed to retrieve credentials"),
+			expectedErr: errors.New("failed to retrieve AWS credentials: failed to retrieve credentials"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockProvider := &mockCredentialsProvider{
+				creds: tt.creds,
+				err:   tt.err,
+			}
+
+			creds, err := getSessionCredentials(context.Background(), mockProvider)
+			if tt.expectedErr != nil {
+				assert.Nil(t, creds)
+				assert.EqualError(t, err, tt.expectedErr.Error())
```

Comment:
This line is also a byproduct of the issue above where each unit test cannot configure the function, but rather, the "expected error message".

This is also a very, very bad idea because it will be comparing strings to strings: if someone modifies your code and changes just the casing of the string (from lowercase to uppercase for example), now none of your tests will pass.